cpp/src/arrow/tensor.cc

```diff
@@ -123,6 +123,100 @@ Type::type Tensor::type_id() const { return type_->id(); }
 
 bool Tensor::Equals(const Tensor& other) const { return TensorEquals(*this, other); }
 
+namespace internal {
```

Comment:
Make this an anonymous namespace `namespace {` as the functions are only used in this `.cc` and should not be visible from anywhere else.

Reply:
Fixed in 5250200